client/engine/store/mockstore.go

```diff
@@ -136,13 +137,99 @@ func (ms MockStore) GetTwoPartyLedger(firstParty types.Address, secondParty type
 
 func (ms MockStore) GetObjectiveByChannelId(channelId types.Destination) (protocols.Objective, bool) {
 	// todo: locking
-	for _, obj := range ms.objectives {
+	for id, objJSON := range ms.objectives {
+		obj, err := decodeObjective(id, objJSON)
+
+		if err != nil {
+			return nil, false
+		}
+
 		for _, ch := range obj.Channels() {
 			if ch.Id == channelId {
+				obj, err = ms.populateChannelData(obj)
+
+				if err != nil {
+					return nil, false // todo: enrich w/ err return
+				}
+
 				return obj, true
 			}
 		}
 	}
 
 	return nil, false
 }
+
+// populateChannelData fetches stored Channel data relevent to the given
+// objective, attaches it to the objective, and returns the objective
+func (ms MockStore) populateChannelData(obj protocols.Objective) (protocols.Objective, error) {
```

Comment:
Right now `populateChannelData` accepts a pointer to an `Objective` and then modifies the objective. The objective pointer that is returned is the same one that is passed in, which could be confusing.

We might want to either:
- Don't bother returning an `Objective` as a return type. `populateChannelData` can just modify the objective in-place.
- Clone the incoming objective and return a completely new objective that is populated with channel data.

Reply:
~(As I read it) `populateChannelData` takes an `Objective`, not a pointer, and so it creates a local copy of the passed objective rather than modifying the original.~

(I believe I've just learned that functions with interface typed parameters always receive them by reference, rather than by value.)

Either way, your specific point still applies, because the pieces of the objective being written to (the channels), *are* pointers. `populateChannelData` can (and does) modify the contents of these pointers in-place.

I'll drop the returned objective since:
- internally cloning here would require modifying the API of virtualfund.Objective, directfund.Objective to make them public
- there is no danger here of multiple threads etc overwriting one another's objective data, since the modified objective itself has just been constructed from JSON

Reply:
addressed in dfb67b6